src/sentry/event_manager.py

```diff
@@ -2007,6 +2007,14 @@ def _save_aggregate_performance(jobs: Sequence[Performance_Job], projects):
             all_group_hashes = [problem.fingerprint for problem in performance_problems]
             group_hashes = all_group_hashes[:MAX_GROUPS]
 
+            if len(group_hashes) > 0:
+                job["event"].data["performance_detector_data"] = {}
+
+            for hash in group_hashes:
+                job["event"].data["performance_detector_data"][
+                    hash
+                ] = performance_problems_by_fingerprint[hash].to_json()
+
```

Comment:
this would save performance_detector_data for new groups that might not get created because of rate limiting. maybe that not an issue

Reply:
Yeah, I think it's better if we only store this for groups we actually create

Reply:
🤔 hmm, right right, I guess I need to use `[...existing_grouphashes, ...new_grouphashes)[: granted_quota.granted]]`, right?

Reply:
I'd follow the same pattern we follow here for adding `groups` to the event